src/reactviews/pages/QueryResult/table/plugins/autoColumnSize.plugin.ts

```diff
@@ -0,0 +1,376 @@
+        let origCols = this._grid.getColumns();
+        let allColumns = deepClone(origCols);
+        allColumns.forEach((col, index) => {
+            col.formatter = origCols[index].formatter;
+            col.asyncPostRender = origCols[index].asyncPostRender;
+        });
+        let column = allColumns[colIndex];
+
+        let autoSizeWidth =
+            Math.max(
+                headerWidth,
+                this.getMaxColumnTextWidth(columnDef, colIndex),
+            ) + 1;
+
+        // Only resize if the current width is smaller than the new width.
+        if (autoSizeWidth > column.width!) {
+            allColumns[colIndex].width = autoSizeWidth;
+            this._grid.setColumns(allColumns);
+            this._grid.onColumnsResized.notify();
+        }
+    }
+
+    /**
+     * For each column, find the max width of the texts in the first MAX_ROW_TO_SCAN rows.
+     * @param columnDefs Column definitions of all columns that need to be resized
+     * @param colIndices Column indices of all columns that need to be resized
+     * @returns An array of the max widths of each column
+     */
+    private getMaxColumnTextWidths(
+        columnDefs: Slick.Column<T>[],
+        colIndices: number[],
+    ): number[] {
+        let data = this._grid.getData() as Slick.DataProvider<T>;
+        let viewPort = this._grid.getViewport();
+        let start = Math.max(0, viewPort.top);
+        let end = Math.min(data.getLength(), viewPort.bottom);
+        let allTexts: Array<string>[] = [];
+        let rowElements: JQuery[] = [];
+
+        columnDefs.forEach((columnDef) => {
+            let texts: Array<string> = [];
+            for (let i = start; i < end; i++) {
+                texts.push(data.getItem(i)[columnDef.field!]);
+            }
+            allTexts.push(texts);
+            let rowEl = this.createRow();
+            rowElements.push(rowEl);
+        });
+
+        let templates = this.getMaxTextTemplates(
+            allTexts,
+            columnDefs,
+            colIndices,
+            data,
+            rowElements,
+        );
+
+        let widths = this.getTemplateWidths(rowElements, templates);
+        rowElements.forEach((rowElement) => {
+            this.deleteRow(rowElement);
+        });
+        if (this._options.maxWidth) {
+            return widths.map((width) =>
+                Math.min(this._options.maxWidth!, width),
+            );
+        } else {
+            return widths.map((width) => width);
+        }
+    }
+
+    private getMaxColumnTextWidth(
+        columnDef: Slick.Column<T>,
+        colIndex: number,
+    ): number {
+        let texts: Array<string> = [];
+        let rowEl = this.createRow();
+        let data = this._grid.getData() as Slick.DataProvider<T>;
+        let viewPort = this._grid.getViewport();
+        let start = Math.max(0, viewPort.top);
+        let end = Math.min(data.getLength(), viewPort.bottom);
```

Comment:
are we going through all the data in the result set, or the visible viewPort? I'm not sure what viewPort.bottom is.  But we probably shouldn't go through the whole result set, maybe 100 or 1000 rows is totally fine to get a general shape of the data.

Reply:
It just goes through the visible viewPort

Reply:
Recording here to make sure it doesn't get lost: the suggestion from sync yesterday was to calculate the width using the first 50-100 rows of data, rather than just the viewport since the viewport is possibly only a small slice.

Reply:
@kburtram @Benjin increased it to 50 rows (don't want to make the calculation too expensive).